Dockerfile

```diff
@@ -0,0 +1,51 @@
+FROM docker.mirror.hashicorp.services/alpine:latest as dev
+
+RUN addgroup vault && \
+    adduser -S -G vault vault
+
+ADD dist/vault-k8s /vault-k8s
+
+USER vault
+
+ENTRYPOINT ["/vault-k8s"]
+
+# This target creates a production release image for the project.
+FROM docker.mirror.hashicorp.services/alpine:latest as default
+
+# PRODUCT_VERSION is the tag built, e.g. v0.1.0
+# PRODUCT_REVISION is the git hash built
+ARG PRODUCT_VERSION
+ARG PRODUCT_REVISION
+ARG PRODUCT_NAME=vault-k8s
+ARG TARGETOS TARGETARCH
+
+# Additional metadata labels used by container registries, platforms
+# and certification scanners.
+LABEL name="Vault K8s" \
+      maintainer="Vault Team <vault@hashicorp.com>" \
+      vendor="HashiCorp" \
+      version=$PRODUCT_VERSION \
+      release=$PRODUCT_VERSION \
+      revision=$PRODUCT_REVISION \
+      summary="The Vault-K8s binary includes first-class integrations between Vault and Kubernetes." \
+      description="Vault-K8s includes first-class integrations between Vault and Kuberentes. Integrations include the Vault Agent Injector mutating admission webhook."
+
+COPY LICENSE /licenses/mozilla.txt
+
+# Create a non-root user to run the software.
+RUN addgroup vault && \
+    adduser -S -G vault vault
+
+# Set up certificates, base tools, and software.
+RUN set -eux && \
+    apk add --no-cache ca-certificates libcap su-exec iputils
+
+COPY dist/$TARGETOS/$TARGETARCH/vault-k8s /bin/
```

Comment:
What's putting the binary in this directory structure? I see the build step zip's the binary into a file:

https://github.com/hashicorp/vault-k8s/pull/324/files#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721R87-R90

But that doesn't match this directory structure.

Reply:
tbh I couldn't have given you a full answer until I looked it up just now, but the short answer is that the `hashicorp/actions-docker-build` action we use extracts the built GitHub artifact into this location.

A digest_inputs script gets run first: https://github.com/hashicorp/actions-docker-build/blob/5f52c4cae49d2f379cd1fcb4a4f09b55f97a3f06/action.yml#L109

That sets a ZIP_LOCATION env var which matches our COPY directory here: https://github.com/hashicorp/actions-docker-build/blob/main/scripts/digest_inputs#L78

We then download the zip artifact into that folder: https://github.com/hashicorp/actions-docker-build/blob/main/action.yml#L119

And finally we just extract the binary into the same directory as the zip: https://github.com/hashicorp/actions-docker-build/blob/main/scripts/extract_product_zip#L13

For reference, that -j option on unzip:

> junk paths. The archive's directory structure is not recreated; all files are deposited in the extraction directory (by default, the current one).